### ISSUE TITLE
PG_CONFIG ./configure environment variable doesn't seem to work

### ISSUE THREAD

**Issue by fdr** [pgbackrest/pgbackrest#1323]
On centos 8, when checking out https://github.com/pgbackrest/pgbackrest/pull/1320 I had to stumble around to build a little:

```
$ ./configure
checking for gcc... gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether gcc accepts -g... yes
checking for gcc option to accept ISO C89... none needed
checking for gcc option to accept ISO C99... none needed
checking build system type... x86_64-pc-linux-gnu
checking host system type... x86_64-pc-linux-gnu
checking whether C compiler accepts -Wno-clobbered... yes
checking for pg_config... no
checking for PQconnectdb in -lpq... no
configure: error: library 'pq' is required
```
Huh, I thought, libpq is installed, maybe it doesn't know where it is:

```
$ ./configure --help | fgrep -B9 PG_CON

Some influential environment variables:
  CC          C compiler command
  CFLAGS      C compiler flags
  LDFLAGS     linker flags, e.g. -L<lib dir> if you have libraries in a
              nonstandard directory <lib dir>
  LIBS        libraries to pass to the linker, e.g. -l<library>
  CPPFLAGS    (Objective) C/C++ preprocessor flags, e.g. -I<include dir> if
              you have headers in a nonstandard directory <include dir>
  PG_CONFIG   path to pg_config utility
```

Alright, let's try that:

```
$ PG_CONFIG=/usr/pgsql-13/bin/pg_config ./configure --prefix=/usr
checking for gcc... gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether gcc accepts -g... yes
checking for gcc option to accept ISO C89... none needed
checking for gcc option to accept ISO C99... none needed
checking build system type... x86_64-pc-linux-gnu
checking host system type... x86_64-pc-linux-gnu
checking whether C compiler accepts -Wno-clobbered... yes
checking for /usr/pgsql-13/bin/pg_config... no
checking for PQconnectdb in -lpq... no
configure: error: library 'pq' is required
```

Nope, same problem. How about tweaking $PATH?

```
$ PATH=$PATH:/usr/pgsql-13/bin ./configure --prefix=/usr
[...]
checking for pg_config... yes
checking for PQconnectdb in -lpq... yes
[....etc, success.]
```

So I concluded the environment variable `PG_CONFIG` isn't working quite right.

**Comment by dwsteele** [comment]
Yeah, this started happening on Fedora at some point and then RHEL/CentOS 8.

I was meaning to look into this and I managed to forget, so thanks for the heads up.

I think it has to do with our order of adding compiler flags. We use a lot of strict compile flags that break the libpq headers. So, if you look the the configure log I think you'll see a compiler error, not a missing lib. But an error at that stage makes configure say the lib is missing.

Can you confirm that there is a compiler error in your `config.log`?

**Comment by dwsteele** [comment]
Ah, never mind. There are some warnings in the log but it's definitely linker error causing the issue.

**Comment by fdr** [comment]
Yeah, PATH works fine, so, no blocking but thought I'd itemize.

**Comment by dwsteele** [comment]
Yeah, but I have a Fedora 32 VM where PATH does not work, even though it does manage to load pg_config and grab settings.

If I start fresh from RHEL8 it works fine:
```
yum install -y git make gcc postgresql-devel openssl-devel libxml2-devel lz4-devel libzstd-devel bzip2-devel
git clone https://github.com/pgbackrest/pgbackrest.git
cd pgbackrest/src
./configure
```
And if I specify a bogus PG_CONFIG I see it:
```
> ./configure PG_CONFIG=/usr/pgsql-10/bin/pg_configs
...
checking for /usr/pgsql-10/bin/pg_configs... no
...
```
But on an affected Fedora 32 build I see:
```
/home/vagrant/pgbackrest/src/configure PG_CONFIG=/usr/pgsql-12/bin/pg_config
...
checking for pg_config... no
...
```
Which is what you saw. The Fedora 32 build is our test container so it has tons of stuff in it. Something is interfering but I don't know what.

The community packages use PG_CONFIG and that works fine.

So basically I'm stumped. I've been looking at this on an off for most of a year and I'm no closer to a solution. The only thing I can think to do is bisect our Fedora 32 test container and see where it starts breaking.

**Comment by fdr** [comment]
Interesting. It does not work fine for me on this CentOS i8 mage without additional options for pg_config. There's no much going on in this image. Would you like to have a look?

**Comment by dwsteele** [comment]
We have been solving this by setting `PATH=/usr/pgsql-{[pg-version]}/bin:$PATH`. This seems to allow configure to work on all RHEL variants. If there is more than one PG version installed we set this to the latest version.
